src/thread.cpp

```diff
@@ -27,7 +27,8 @@
 
 using namespace Search;
 
-ThreadPool Threads; // Global object
+ThreadPool Threads;
+thread_local Thread *ThisThread;
```

Comment:
See here:
https://software.intel.com/en-us/blogs/2011/05/02/the-hidden-performance-cost-of-accessing-thread-local-variables

"In order to understand where the cost of accessing thread local variable comes from, I needed to understand how a compiler implements it. The compiler will assign a unique global ID to each thread-local variable (this ID is the same for each thread), and maintains a Thread Local Storage (TLS) lookup table for each thread. The global ID then used to find the address of any thread local variable. So the cost of using a thread-local variable includes the cost of a function call and a lookup in the indexed table. "